library/lua/3rdparty/alt_getopt.lua

```diff
@@ -80,14 +88,26 @@ local function canonicalize(options, opt)
                 'Option "%s" resolves to non-number for has_arg flag', opt))
     end
 
+    ---@type integer
     return opt
 end
 
+---@nodiscard
+---@param options table<string, string|integer>
+---@param opt string|integer
+---@return boolean
 local function has_arg(options, opt)
     return options[canonicalize(options, opt)] == 1
 end
 
 -- returns vectors for opts, optargs, and nonoptions
+---@nodiscard
+---@param args string[] e.g, { ... }
+---@param sh_opts string e.g., 'ak:hv'
+---@param long_opts table<string, string|integer>
```

Comment:
isn't this a table<string, string>?

Reply:
`processArgsGetopt` assigns a truthy 1 or 0 to it. So I guess it kind of like the other "just has to be a truthy" fields.